datafusion/physical-expr/src/planner.rs

```diff
@@ -393,7 +393,10 @@ pub fn create_physical_expr(
                     execution_props,
                 )?);
             }
-
+            // udfs with zero params expect null array as input
+            if args.is_empty() {
+                physical_args.push(Arc::new(Literal::new(ScalarValue::Null)));
```

Comment:
Though a convenient implementation, I'm wary that generating a fake argument by which to transfer row count will cause problems later.

I'd prefer to see a solution where the row count is passed out-of-band, for example in a change to https://github.com/apache/arrow-datafusion/blob/cef119da9ee8672b1b1e50ac01387dcb1640d96e/datafusion/expr/src/function.rs#L39 that would add an extra argument (i.e. `len: usize`) for this purpose.

If that were present, we could populate it up in the call stack where we know the RecordBatch size, probably here: https://github.com/apache/arrow-datafusion/blob/cef119da9ee8672b1b1e50ac01387dcb1640d96e/datafusion/physical-expr/src/scalar_function.rs#L147

Reply:
Sounds good. Thanks for the suggestions!

Reply:
Oh, sorry I missed that. I didn't realize this was part of the existing design.

Reply:
No worries I should've put that in the PR itself. Just edited